client/my-sites/upgrades/domain-management/edit/card/header/index.jsx

```diff
@@ -21,16 +24,15 @@ const Header = React.createClass( {
 	},
 
 	render() {
-		const domain = this.props.domain;
+		const { domain, isDomainOnly } = this.props;
 
-		if ( ! domain ) {
+		if ( ! domain || isDomainOnly ) {
```

Comment:
As discussed on Slack, this isn't really specific to domain-only sites: the header here is redundant for any domain. What do you think about either just hiding the flag in this case or removing the header entirely?

cc @ranh who discussed this in the issue